packages/perspective-viewer-d3fc/src/js/zoom/zoomableChart.js

```diff
@@ -37,12 +38,27 @@ export default () => {
 
                 const noZoom = transform.k === 1 && transform.x === 0 && transform.y === 0;
 
-                getZoomControls(selection)
-                    .style("display", noZoom ? "none" : "")
-                    .select("#zoom-reset")
-                    .on("click", () => {
-                        selection.select(chartPlotArea).call(zoom.transform, d3.zoomIdentity);
+                const zoomControls = getZoomControls(selection).style("display", noZoom ? "none" : "");
+                zoomControls.select("#zoom-reset").on("click", () => selection.select(chartPlotArea).call(zoom.transform, d3.zoomIdentity));
+
+                const oneYear = zoomControls.select("#one-year").style("display", dateAxis ? "" : "none");
+                if (dateAxis) {
+                    const xRange = xCopy.range();
+                    oneYear.on("click", () => {
+                        const start = new Date(xScale.domain()[0]);
+                        const end = new Date(start);

```

Comment:
Would this look better if it zoomed to the middle? i.e. work out the middle of the current domain then do half a year either side of that?

Reply:
I can certainly take a look.